Made `LaunchConfiguration` asynchronous at the language level

Lehonti opened the pull request:
Refactoring `AsyncEffectRenderer` properly is currently hard because it's hard to track when exactly things are happening.

For instance, methods where configuration dialogs are created return `void` and rely on event handlers for the dialog's asynchronous tasks, most importantly what happens when the user clicks either `Ok` or `Cancel`. This leaves these dialogs 'hanging'.

The above is what I'm addressing in this pull request, but there are many other things that I plan to address in future pull requests.

cameronwhite commented on Pinta.Core/Effects/BaseEffect.cs:
```diff
@@ -84,35 +84,12 @@ public abstract class BaseEffect
-	/// The localizer for the effect add-in. This is used to fetch translations for the
-	/// strings in the dialog.
-	/// </param>
-	protected void LaunchSimpleEffectDialog (AddinLocalizer localizer)
```
I'm pretty sure this signature (or some equivalent way of doing this) is still needed for addins, since this wrapper translates the `Mono.Addins.AddinLocalizer` into the `IAddinLocalizer` interface
Addins can't create an `AddinLocalizerWrapper` manually either since that's an internal class.

e.g. https://github.com/PintaProject/NightVisionEffect/blob/a9ac90a38373cec4c5f69d042998c4ffe35e6438/NightVisionAddin/NightVisionEffect.cs#L53
and 
168115e75d0cc05338d18354439d22e518ffb00b

Lehonti replied:
Thank you. I've re-added it, but the return type is a `Task`. Is that fine?

cameronwhite replied:
Yeah changing the return type to `Task<bool>` is fine - all the demo add-ins will need to be updated for any API changes before release

Lehonti replied:
@cameronwhite I know that you are referring to API changes in general (in all the extensible parts of Pinta). With that being said, is there another demo _effect_ besides the night vision one?

cameronwhite replied:
No, I think that's the only demo for an effect

cameronwhite commented on Pinta.Core/Effects/BaseEffect.cs:
```diff
@@ -84,35 +84,12 @@ public abstract class BaseEffect
 	/// Launches the configuration dialog for this effect/adjustment.
 	/// If IsConfigurable is true, the ConfigDialogResponse event will be invoked when the user accepts or cancels the dialog.
 	/// </summary>
-	public virtual void LaunchConfiguration ()
```
What do you think about making the return type a `Task<bool>`, or some wrapper around a bool to make the meaning more clear, sort of like  the `ConfigDialogResponseEventArgs` from before?

My main thoughts are
- `Gtk.ResponseType` has a lot of values in the enum, so it's less clear which values we care about or are possible to have to deal with. It's really just success / failure that matter
- Ideally the effect API should be as independent as possible from UI stuff like GTK. Using the `ResponseType` enum probably doesn't cause anything bad like forcing GTK libraries to be loaded / initialized for the unit tests, but it does mean that any API changes when upgrading to future GTK versions have more of an impact on the effects

Lehonti replied:
I think `Task<bool>` is good for now while the API is being defined. I have changed it now

cameronwhite reviewed (commented):
I think this is the right approach. The API actually used to return a `bool` back in the GTK3 days when the dialog was blocking, but then these event handlers were introduced in the GTK4 port (ef0466d5128024ae302f63e84b42e2c435e16bff) which happened before we got the synchronization contexts and async methods working properly

Lehonti commented:
I addressed your points and I also changed `ErrorDialog.ShowMessage` to return a `Task`. I didn't change `ShowError` because it's a little bit more involved and it might be better if I do it in a future pull request

cameronwhite commented:
Thanks, yeah I think limiting this PR to the effect dialog changes makes sense.
For changing other dialogs, note that we do have a `RunAsync()` extension method for `Adw.MessageDialog` so that could be more widely used
